File: app/src/main/java/io/github/tobyhs/weatherweight/AppComponent.java
package io.github.tobyhs.weatherweight;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.AndroidInjector;

import io.github.tobyhs.weatherweight.forecast.ForecastModule;

/**
 * Dagger component for the application instance
 */
@Singleton
@Component(
        modules = {
                AndroidInjectionModule.class,
                AppModule.class,
                ForecastModule.class,
        }
)
public interface AppComponent extends AndroidInjector<App> {
    @Component.Factory
    interface Factory {
        AppComponent create(AppModule appModule);
    }
}

File: app/src/main/java/io/github/tobyhs/weatherweight/AppComponent.kt
package io.github.tobyhs.weatherweight

import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector

import io.github.tobyhs.weatherweight.forecast.ForecastModule

import javax.inject.Singleton

/**
 * Dagger component for the application instance
 */
@Singleton
@Component(modules = [AndroidInjectionModule::class, AppModule::class, ForecastModule::class])
interface AppComponent : AndroidInjector<App> {
    @Component.Factory
    interface Factory {
        fun create(appModule: AppModule): AppComponent
    }
}
